### PR TITLE
Handling mentions

### PR DESCRIPTION
**Pull request by jane-leon**
### Summary 
  Added a searchable dropdown feature for selecting shoutout recipients (with tags) and a toggle to view shoutouts that
  mention you.

  Changes:
  1. Searchable Recipient Selection
    - Replaced text input with a searchable dropdown/autocomplete
    - As you type in the "Who is awesome?" field, the member list filters 
    - Filters by first name, last name, or full name
    - Click to select from the filtered results
  2. Shoutout Mentions View
    - Added toggle button to switch between "Given Shoutouts" and "Shoutouts Mentioning You" views
    - Implemented client side filtering to detect mentions of user's name in receiver field
    - Fetches all shoutouts and filters for mentions containing user's first name, last name, or full name

### Test Plan 

  1. Navigate to the shoutouts page
  2. Test recipient selection:
    - Start typing a name in "Who is awesome?" field
    - Verify the dropdown appears with filtered members
    - Verify clicking a member gives a tag
    - Verify the dropdown closes after selection
  3. Test mentions toggle:
    - Verify toggle buttons ("Given" and "Mentions") appear next to the heading
    - Click "Given" - should show shoutouts you've given to others
    - Click "Mentions" - should show shoutouts where your name appears in the receiver field
  4. Test edge cases:
    - Shoutouts with just a first name (casing shouldn't matter)
    - Shoutouts with full name
    - Shoutouts mentioning multiple people
    - No shoutouts

### Notes 
The searchable dropdown ensures consistent name formatting and eliminates typos

### CODE REVIEW

**Review by ladriennel — CHANGES_REQUESTED** [review]
Hi Jane! Thanks for working on this and great work so far on this PR. I'm just gonna leave a few comments here on potential fixes and suggestions!

**Review by ladriennel — COMMENTED** [review]
One thing I noticed was that in the mentions tab, users can edit and delete shoutouts which they didn’t write. One potential fix could be adding a `canEdit` field to the `ShoutoutCard` component that checks if the user is the giver of that shoutout.

File: frontend/src/components/Forms/ShoutoutsPage/ShoutoutsPage.tsx [hunk 1]
```diff
@@ -1,16 +1,23 @@
...
+import { Message, Loader, Button } from 'semantic-ui-react';
 import { useUserEmail } from '../../Common/UserProvider/UserProvider';
+import { useMembers } from '../../Common/FirestoreDataProvider';
 import { Emitters } from '../../../utils';
 import ShoutoutForm from './ShoutoutForm';
 import ShoutoutList from './ShoutoutList';
 import styles from './ShoutoutsPage.module.css';
 import ShoutoutsAPI from '../../../API/ShoutoutsAPI';
 
+type ShoutoutView = 'given' | 'mentions';
```
Comment (shah-esha) [inline frontend/src/components/Forms/ShoutoutsPage/ShoutoutsPage.tsx]: This is more nitpicky than anything, but I think it might be more intuitive to change "Mentions" to "Received."

File: frontend/src/components/Forms/ShoutoutsPage/ShoutoutsPage.tsx [hunk 2]
```diff
@@ -29,9 +36,56 @@ const ShoutoutsPage: React.FC = () => {
       });
   }, [userEmail]);
 
+  const getMentionShoutouts = useCallback(() => {
```
Comment (shah-esha) [inline frontend/src/components/Forms/ShoutoutsPage/ShoutoutsPage.tsx]: I also noticed that the Mentions don't update after submitting a shoutout (you have to refresh the page), unlike the Given shoutouts, which update immediately after you submit the shoutout Form. This is likely because of this piece of code: 

`getGivenShoutouts()` - (line 98 of the ShoutoutForm.tsx)

I don't think this is a major issue, but it might be nice to have the Mentions also update in real time, and something we can think about in the future.

Some possible solutions:

1. After a user submits the form, only getGivenShoutouts() is updated. Calling getMentionShoutouts() might provide a good temporary fix, as a member's Mentions will then update at least every time they submit a shoutout.
2. I noticed that there is an Emitter for successful form submissions. Another, maybe more robust but harder to implement, solution would be to have a listener for that Emitter, which then updates the respective receiver's list of Mentions.

File: frontend/src/components/Forms/ShoutoutsPage/ShoutoutsPage.tsx [hunk 3]
```diff
@@ -40,16 +94,28 @@ const ShoutoutsPage: React.FC = () => {
...
-        ) : givenShoutouts.length > 0 ? (
+        ) : displayedShoutouts.length > 0 ? (
           <ShoutoutList
-            shoutouts={givenShoutouts.sort((a, b) => a.timestamp - b.timestamp)}
+            shoutouts={displayedShoutouts.sort((a, b) => a.timestamp - b.timestamp)}
             setGivenShoutouts={setGivenShoutouts}
           />
         ) : (
-          <Message>Give someone a shoutout!</Message>
+          <Message>{emptyMessage}</Message>
```
Comment (shah-esha) [inline frontend/src/components/Forms/ShoutoutsPage/ShoutoutsPage.tsx]: I also noticed that when I add a tag or type in the shoutout text, the form grows and overlaps with the Given/Mentions section and covers the “Send” button. Might be a CSS issue (maybe a fixed height or missing `height: auto` in ShoutoutForm.module.css or ShoutoutsPage.module.css).

Also, I noticed that it was possible to pass in "received" as an argument in `getShoutouts()` in the shoutoutAPIs file. Were you able to see what it does?
